### PR TITLE
add: SICK-BR

### PR DESCRIPTION
**Pull request by mrshu**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

Portugese version of the SICK dataset as an STS task, which I believe is a language that is not currently covered by STS.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [ ] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by mrshu** [comment]
One thing I am unsure of is how to deal with the fact that the dataset is a bit larger (4096 examples in the `test` subset). Should it be subsampled, eventhough other SICK datasets in `mteb` are not?

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            "revision": "0cdfb1d51ef339011c067688a3b75b82f927c097",
+        },
+        description="SICK-BR is a Portuguese inference corpus, aligned to and translated from SICK",
+        reference="https://huggingface.co/datasets/eduagarcia/sick-br",
+        type="STS",
+        category="s2s",
+        eval_splits=["validation", "test"],
+        eval_langs=["por-Latn"],
+        main_score="cosine_spearman",
+        date=None,
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:22]: Make sure to fill these out, tests won't pass otherwise

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class SickBrSTS(AbsTaskSTS):
+    metadata = TaskMetadata(
+        name="SICK-BR",
+        dataset={
+            "path": "eduagarcia/sick-br",
+            "revision": "0cdfb1d51ef339011c067688a3b75b82f927c097",
+        },
+        description="SICK-BR is a Portuguese inference corpus, aligned to and translated from SICK",
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:15]: I have a feeling we are not that fond of having translated datasets (right, @KennethEnevoldsen ?), is there a particular reason this dataset should make an exception.

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.rename_columns(
+            {
+                "sentence_A": "sentence1",
+                "sentence_B": "sentence2",
+                "relatedness_score": "score",
+                "pair_ID": "id",
+            }
+        )
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:51]: I believe subsampling would be nice here. We are currently making efforts to make the benchmark run faster and I highly doubt having 2000 examples would seriously  change scores.

**Review by x-tabdeveloping — CHANGES_REQUESTED** [review]
Doesn't look bad, though I'm a bit doubtful whether it's a good idea to include translated datasets. You can also see that the performance change is negligible between the two models. Changes are definitely needed before merging, and let's discuss whether we should include this in the benchmark

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 1 above)
Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:22]: Thanks @x-tabdeveloping.

Interestingly, some existing datasets don't seem to have these either: https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/STS/pol/PolishSTS.py#L21-L30

Comment (x-tabdeveloping) [reply mteb/tasks/STS/por/SickBrSTS.py:22]: Yes, I know :D It's an issue we're working on. If you look at the points table, we also offer points for filling in metadata on older tasks.
We would just like new tasks to include all metadata if possible.

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 2 above)
Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: @x-tabdeveloping sorry, I did not realize that was the case.

Feel free to close this PR then.

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 1 above)
Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:22]: Understood, thanks @x-tabdeveloping!

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 2 above)
Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: That said, you should remove the SickrPLSTS (https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/STS/pol/PolishSTS.py#L7) as well then -- here is what it's paper (https://aclanthology.org/2020.lrec-1.207.pdf) says:

> We share a manually translated version of SICK (Sentences Involving Compositional Knowledge) dataset,
containing 10,000 pairs of sentences annotated with
textual entailment labels and semantic relatedness
scores.

Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: @KennethEnevoldsen I would greatly appreciate if you could share your thoughts here so that I can either finish this up or close it altogether?

Thanks!

**Review by KranthiGV — CHANGES_REQUESTED** [review]
Failing tests are due to missing metadata. Can you fix it?

**Comment by mrshu** [comment]
Sure thing, as long as the dataset is desired, I'd be very happy to fill it all out!

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 2 above)
Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: Understood, thanks @KennethEnevoldsen. I believe this particular dataset was only bootstrapped from English translation -- it was then manually checked, even twice in some cases.

This is from Section 3.2 of the associated paper (https://linux.ime.usp.br/~thalen/SICK_PT.pdf):

> The 6076 unique sentences of the corpus were equally distributed among the
10 annotators. We used an online platform for the checking. This made it possible
for annotators to look at each others’ work when translating their sentences. We
also kept an online forum for discussing issues, where more than 2k messages were
exchanged during this work. The glossary was always updated when a solution
was reached. Each annotator could also mark out complex sentences that they
thought needed further review. Differently from other corpora creation processes
we know about, our annotators could always say they were not able to annotate
something, an easy strategy that helps to ensure the quality of the work. Finally,
an experienced annotator double checked all sentences considered complex and
proposed a final translation for these sentences.

Would you find that to be enough for the dataset to be considered?

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        main_score="cosine_spearman",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/STS/por/SickBrSTS.py:30]: should be machine translated and then validated

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: Ahh, so it is human-translated. I think this is a fine dataset to add.

Comment (KennethEnevoldsen) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: I would also make this clear in the description.

**Comment by x-tabdeveloping** [comment]
@KranthiGV The author has already been alerted by me about the metadata issues. I believe it's a bit confusing and unnecessary to have two reviewers on the same PR. Can you please review PRs that do not have a reviewer already assigned to them (and of course assign yourself so others can see you're handling it)?

**Comment by KranthiGV** [comment]
> @KranthiGV The author has already been alerted by me about the metadata issues. I believe it's a bit confusing and unnecessary to have two reviewers on the same PR. Can you please review PRs that do not have a reviewer already assigned to them (and of course assign yourself so others can see you're handling it)?

Sure, thanks! Sorry for the confusion.

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 2 above)
Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:15]: Understood, thank you @KennethEnevoldsen, will certainly make it so!

File: mteb/tasks/STS/por/SickBrSTS.py (hunk 3 above)
Comment (x-tabdeveloping) [reply mteb/tasks/STS/por/SickBrSTS.py:51]: I would just do a train-test split (either with Huggingface or with sklearn) on the test set and only keep one half. Some other tasks do have code like this, if you have a hard time implementing it, I can write you some code bits tomorrow.

Comment (mrshu) [reply mteb/tasks/STS/por/SickBrSTS.py:82]: Understood, thank you @x-tabdeveloping -- I went with a shuffle and a straightforward `select` after that. I'd appreciate if you could take a look.

**Comment by mrshu** [comment]
@KennethEnevoldsen @x-tabdeveloping @KranthiGV I would appreciate if you could take another look. Thanks!

File: mteb/tasks/PairClassification/por/SickBrPC.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  booktitle="Computational Processing of the Portuguese Language. PROPOR 2018.",
+  doi ="10.1007/978-3-319-99722-3_31",
+  isbn="978-3-319-99722-3"
+}
+        """,
+        n_samples={"validation": N_SAMPLES, "test": N_SAMPLES},
+        avg_character_length={"validation": N_SAMPLES, "test": 54.89},
+    )
+
+    def dataset_transform(self):
```
Comment (x-tabdeveloping) [inline mteb/tasks/PairClassification/por/SickBrPC.py:57]: This is a bit too much code for my taste.
The subsampling and shuffling stuff could easily be taken care of with `self.stratified_subsampling`.
I also fail to see why you would filter out neutral labels. Is that not something we want to be able to detect?
Here's a snippet that I would prefer:
```python
def label_to_num(row: dict) -> dict:
    label = row["entailment_label"]
    if label == "NEUTRAL":
        row["labels"] = 0
    if label == "ENTAILMENT":
        row["labels"] = 1
    if label == "CONTRADICTION":
        row["labels"] = 2
    return row

def dataset_transform(self):
    self.dataset = self.stratified_subsampling(self.dataset, seed=self.seed, splits=self.metadata.eval_splits, label="label", n_samples=N_SAMPLES)
    self.dataset = self.dataset.map(label_to_num)
    self.dataset = self.dataset.rename_columns({"sentence_A": "sent1", "sentence_B": "sent2"})
```

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        name="SICK-BR-STS",
+        dataset={
+            "path": "eduagarcia/sick-br",
+            "revision": "0cdfb1d51ef339011c067688a3b75b82f927c097",
+        },
+        description="SICK-BR is a Portuguese inference corpus, human translated from SICK",
+        reference="https://linux.ime.usp.br/~thalen/SICK_PT.pdf",
+        type="STS",
+        category="s2s",
+        eval_splits=["validation", "test"],
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:21]: Same.

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        eval_langs=["por-Latn"],
+        main_score="cosine_spearman",
+        date=("2018-01-01", "2018-09-01"),  # rough estimate
+        form=["written"],
+        domains=["Web"],
+        task_subtypes=[],
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:27]: This should be assigned preferably.

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    and de Paiva, Valeria"
+  year ="2018",
+  title="SICK-BR: A Portuguese Corpus for Inference",
+  booktitle="Computational Processing of the Portuguese Language. PROPOR 2018.",
+  doi ="10.1007/978-3-319-99722-3_31",
+  isbn="978-3-319-99722-3"
+}
+        """,
+        n_samples={"validation": N_SAMPLES, "test": N_SAMPLES},
+        avg_character_length={"validation": N_SAMPLES, "test": 54.89},
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:55]: Same

File: mteb/tasks/STS/por/SickBrSTS.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+    @property
+    def metadata_dict(self) -> dict[str, str]:
+        metadata_dict = super().metadata_dict
+        metadata_dict["min_score"] = 1
+        metadata_dict["max_score"] = 5
+        return metadata_dict
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.shuffle(seed=self.seed)
```
Comment (x-tabdeveloping) [inline mteb/tasks/STS/por/SickBrSTS.py:66]: I would just use HF's inbuilt `train_test_split()` method.
Like so:
```python
 for split in self.metadata.eval_splits:
            self.dataset.update(
                {
                    split: self.dataset[split].train_test_split(
                        test_size=N_SAMPLES, seed=self.metadata.seed
                    )["test"]
                }
            )
```

**Review by x-tabdeveloping — CHANGES_REQUESTED** [review]

File: mteb/tasks/PairClassification/por/SickBrPC.py (hunk 5 above)
Comment (mrshu) [reply mteb/tasks/PairClassification/por/SickBrPC.py:57]: Thanks @x-tabdeveloping. I modeled this after the `KlueNLI` and `CTKFactsNLI` which do this very similarly. As far as I understand, the PairClassification task would prefer two classes:

https://github.com/embeddings-benchmark/mteb/blob/2562644af1c1d34edbd8aa378386ac04bec43a40/mteb/tasks/PairClassification/ces/CTKFactsNLI.py#L51-L54

https://github.com/embeddings-benchmark/mteb/blob/2562644af1c1d34edbd8aa378386ac04bec43a40/mteb/tasks/PairClassification/kor/KlueNLI.py#L47-L50

**Comment by mrshu** [comment]
Thanks for the feedback @x-tabdeveloping, I tried to incorporate your feedback and would appreciate if you could take another look.

**Review by KennethEnevoldsen — APPROVED** [review]
I think this is ready to merge as long as @x-tabdeveloping agrees. Feel free to add points as well

File: mteb/tasks/PairClassification/por/SickBrPC.py (hunk 5 above)
Comment (x-tabdeveloping) [reply mteb/tasks/PairClassification/por/SickBrPC.py:57]: Alrght! I will look into this, but it's good for now, I see the rationale.

**Review by x-tabdeveloping — APPROVED** [review]
Looks good now, thanks for the quick fixes!

**Comment by mrshu** [comment]
Thanks for the approval @x-tabdeveloping! Unfortunately, I haven't really added any points here, so I am not sure what would be the best way of doing so. Should I open a new PR?

**Comment by KennethEnevoldsen** [comment]
@mrshu open a new PR with the points (using the number of this PR).

**Comment by x-tabdeveloping** [comment]
Oh sorry for that. I will approve the points as soon as you submit the PR just tag me.

**Comment by mrshu** [comment]
Thanks a bunch everyone, it should be ready in https://github.com/embeddings-benchmark/mteb/pull/754
